torch/_refs/nn/functional/__init__.py

```diff
@@ -1116,7 +1116,6 @@ def relu6(a: TensorLikeType, inplace: bool = False) -> TensorLikeType:
     return torch.nn.functional.hardtanh(a, 0, 6)
 
 
-@register_decomposition(aten.glu)
```

Comment:
What was wrong with this one? It looks the same except it also gets type promotion correct.

Reply:
@peterbell10 nothing wrong with it; it's more that since we are adding this as a core decomposition, we would like to move it under `_decomp/decompositions.py`. Fair point about the type promotion though; we should add that to the moved function as well.

Reply:
I don't think it's great practice to copy-paste decompositions from one part of the codebase to another. This isn't the only case of that: https://github.com/pytorch/pytorch/pull/109319/. 

Some of the other prs don't have tests associated with them (hanh window is not in opinfo, so you don't testing for free) and has perf issues with it: https://github.com/pytorch/pytorch/pull/108723/files#r1330388161

Reply:
@peterbell10 I see, given what you've described I think we should not duplicate the implementation but just use the original one.

@eellison given this discussion thread I also think we should avoid copy-pasting decompositions going forward. Regarding hanh window specifically, what is the best practice for testing if it isn't in opinfo?

Reply:
@eellison I closed that PR for hann_window since it wasn't syncing with my diff properly and made a new PR which does have testing.
https://github.com/pytorch/pytorch/pull/109464

Reply:
@eellison 
> and has perf issues with it:

Which perf issues are you referring to?

Reply:
Adding opinfo inputs as was done in the alternate is good ! The perf issues is instantiating on cpu then copying to cuda, which will block inductor fusion